backend/src/apiserver/client_manager/client_manager.go

```diff
@@ -178,12 +208,61 @@ func (c *ClientManager) init() {
 	// UUID generator
 	c.uuid = util.NewUUIDGenerator()
 
+	var controllerClient ctrlclient.Client
+
+	var pipelineStoreForRef storage.PipelineStoreInterface
+
+	if options.UsePipelineKubernetesStorage {
+		restConfig, err := util.GetKubernetesConfig()
+		if err != nil {
+			panic(fmt.Sprintf("Failed to initialize the RestConfig: %v", err))
+		}
+
+		var cacheConfig map[string]cache.Config
+
+		if !common.IsMultiUserMode() && common.GetPodNamespace() != "" {
+			cacheConfig = map[string]cache.Config{common.GetPodNamespace(): {}}
+		}
+
+		k8sAPICache, err := cache.New(restConfig,
+			cache.Options{
+				DefaultNamespaces: cacheConfig,
+				Scheme:            scheme,
+			},
+		)
+		if err != nil {
+			glog.Errorf("Failed to generate a cache to the cluster: %v", err)
+		}
+
+		go func() {
+			defer options.WaitGroup.Done()
+
+			err := k8sAPICache.Start(options.Context)
+			if err != nil {
+				panic(fmt.Sprintf("Failed to start the cache to the cluster: %v", err))
+			}
+		}()
+
+		controllerClient, err = ctrlclient.New(
+			restConfig, ctrlclient.Options{Scheme: scheme, Cache: &ctrlclient.CacheOptions{Reader: k8sAPICache}},
+		)
+		if err != nil {
+			glog.Errorf("Failed to initialize the client: %v", err)
```

Comment:
I think you misunderstood my comment. Could you return errors in this function rather than panic or just log the errors? Then the calling function such `init` can exit with the error.

Reply:
My apologies to that. I changed the code according to your suggestion now.